MantidQt/MantidWidgets/src/InstrumentView/PanelsSurface.cpp

```diff
@@ -231,9 +219,8 @@ void PanelsSurface::addFlatBank(ComponentID bankId,
     m_unwrappedDetectors.reserve(m_unwrappedDetectors.size() + nelem);
     for (int i = 0; i < nelem; ++i) {
       // setup detector info
-      Mantid::Geometry::IDetector_const_sptr det =
-          boost::dynamic_pointer_cast<const Mantid::Geometry::IDetector>(
-              assembly->getChild(i));
+      auto det = boost::dynamic_pointer_cast<const Mantid::Geometry::IDetector>(
+          assembly->getChild(i));
       if (!doneRotation) {
         pos0 = det->getPos();
```

Comment:
Ok @SimonHeybrock maybe we should look at this as a separate item. I think we have some of this functionality in the prototype already.